erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -1664,7 +1696,7 @@ func (hph *HexPatriciaHashed) fold() (err error) {
 		if hph.trace {
 			fmt.Printf("formed leaf (%d %x, depth=%d) [%x] %s\n", row, nibble, depth, updateKey, cell.FullString())
 		}
-	default: // Branch node
+	case updateKindBranch:
```

Comment:
Still good to have a default case with an error handler.

Reply:
kind could not be of any other value due to function counting 1's in 16 bits: all possible cases handled

Reply:
It's fine in any case, just thought it might be better to cover all cases at a static level, but the linter is not complaining so it's fine either way.